baremetal/metal3machine_manager.go

```diff
@@ -95,10 +95,12 @@ type MachineManager struct {
 
 	Cluster               *capi.Cluster
 	Metal3Cluster         *capm3.Metal3Cluster
+	MachineList           *capi.MachineList
 	Machine               *capi.Machine
 	Metal3Machine         *capm3.Metal3Machine
 	Metal3MachineTemplate *capm3.Metal3MachineTemplate
-	MachineSetList        []*capi.MachineSet
+	MachineSet            *capi.MachineSet
+	MachineSetList        *capi.MachineSetList
```

Comment:
You are removing the array/List `[]`, so is it not a list anymore ?

Reply:
Yes, it is struct and it has MachineSet list field inside